### ISSUE TITLE
Reminder did not close for calendar that repeated for Mutiple days

### ISSUE THREAD

**Issue by bnguye21** [kewisch/gdata-provider#554]
### Describe the problem and steps to reproduce it:
Support you have thunderbird newest version (102.5.1) and Provider for Google Calendar (102.0.3) installed on either windows 10 or windows 11
Add someone in your team to access to their google calendar and create a repeat calendar, invite few people and submit
In your thunderbird, after sync, you will see the calendar.  In reminder, you will see the pop up for the reminder and button for dismiss
Click dismiss but nothing happen 

(Please include as many details as possible.)

### What happened?
reminder did not close after click dismiss 
### What did you expect to happen?
reminder should close

### Anything else we should know?

Calendar: [calGoogleCalendar] Modifying item Weekly touch base for SAODS failed:2147500037: {
 "error": {
  "errors": [
   {
    "domain": "global",
    "reason": "forbidden",
    "message": "Forbidden"
   }
  ],
  "code": 403,
  "message": "Forbidden"
 }
}
[gdataCalendar.jsm:575](resource://gdata-provider/legacy/modules/gdataCalendar.jsm)
Calendar: [calGoogleCalendar] Modifying item Annual Staff Thanksgiving Lunch Meeting-RSVP by November 4 failed:2147500037: {
 "error": {
  "errors": [
   {
    "domain": "calendar",
    "reason": "forbiddenForNonOrganizer",
    "message": "Shared properties can only be changed by the organizer of the event."
   }
  ],
  "code": 403,
  "message": "Shared properties can only be changed by the organizer of the event."
 }
}
[gdataCalendar.jsm:575](resource://gdata-provider/legacy/modules/gdataCalendar.jsm)
Calendar: [calGoogleCalendar] Modifying item Town Hall Planning failed:2147500037: {
 "error": {
  "errors": [
   {
    "domain": "calendar",
    "reason": "forbiddenForNonOrganizer",
    "message": "Shared properties can only be changed by the organizer of the event."
   }
  ],
  "code": 403,
  "message": "Shared properties can only be changed by the organizer of the event."
 }
}
[gdataCalendar.jsm:575](resource://gdata-provider/legacy/modules/gdataCalendar.jsm)
NS_ERROR_FAILURE: [xpconnect wrapped (nsIClassInfo, calIItemBase, calIEvent, calIInternalShallowCopy)] 7 [calCachedCalendar.js:747](resource:///components/calCachedCalendar.js)
Calendar: [calGoogleCalendar] Modifying item Annual Staff Thanksgiving Lunch Meeting-RSVP by November 4 failed:2147500037: {
 "error": {
  "errors": [
   {
    "domain": "calendar",
    "reason": "forbiddenForNonOrganizer",
    "message": "Shared properties can only be changed by the organizer of the event."
   }
  ],
  "code": 403,
  "message": "Shared properties can only be changed by the organizer of the event."
 }
}
[gdataCalendar.jsm:575](resource://gdata-provider/legacy/modules/gdataCalendar.jsm)
NS_ERROR_FAILURE: [xpconnect wrapped (nsIClassInfo, calIItemBase, calIEvent, calIInternalShallowCopy)] [calCachedCalendar.js:747](resource:///components/calCachedCalendar.js)
Calendar: [calGoogleCalendar] Modifying item Town Hall Planning failed:2147500037: {
 "error": {
  "errors": [
   {
    "domain": "calendar",
    "reason": "forbiddenForNonOrganizer",
    "message": "Shared properties can only be changed by the organizer of the event."
   }
  ],
  "code": 403,
  "message": "Shared properties can only be changed by the organizer of the event."
 }
}
2 [gdataCalendar.jsm:575](resource://gdata-provider/legacy/modules/gdataCalendar.jsm)
Calendar: [calGoogleCalendar] Modifying item Town Hall Planning Follow-up failed:2147500037: {
 "error": {
  "errors": [
   {
    "domain": "calendar",
    "reason": "forbiddenForNonOrganizer",
    "message": "Shared properties can only be changed by the organizer of the event."
   }
  ],
  "code": 403,
  "message": "Shared properties can only be changed by the organizer of the event."
 }
}
[gdataCalendar.jsm:575](resource://gdata-provider/legacy/modules/gdataCalendar.jsm)
Calendar: [calGoogleCalendar] Modifying item Weekly touch base for SAODS failed:2147500037: {
 "error": {
  "errors": [
   {
    "domain": "global",
    "reason": "forbidden",
    "message": "Forbidden"
   }
  ],
  "code": 403,
  "message": "Forbidden"
 }
}
[gdataCalendar.jsm:575](resource://gdata-provider/legacy/modules/gdataCalendar.jsm)
NS_ERROR_FAILURE: [xpconnect wrapped (nsIClassInfo, calIItemBase, calIEvent, calIInternalShallowCopy)] 4 [calCachedCalendar.js:747](resource:///components/calCachedCalendar.js)
Calendar: [calGoogleCalendar] Modifying item OneIT Winter Town Hall failed:2147500037: {
 "error": {
  "errors": [
   {
    "domain": "calendar",
    "reason": "forbiddenForNonOrganizer",
    "message": "Shared properties can only be changed by the organizer of the event."
   }
  ],
  "code": 403,
  "message": "Shared properties can only be changed by the organizer of the event."
 }
}
[gdataCalendar.jsm:575](resource://gdata-provider/legacy/modules/gdataCalendar.jsm)
NS_ERROR_FAILURE: [xpconnect wrapped (nsIClassInfo, calIItemBase, calIEvent, calIInternalShallowCopy)] [calCachedCalendar.js:747](resource:///components/calCachedCalendar.js)
Calendar: [calGoogleCalendar] Modifying item Town Hall Planning failed:2147500037: {
 "error": {
  "errors": [
   {
    "domain": "calendar",
    "reason": "forbiddenForNonOrganizer",
    "message": "Shared properties can only be changed by the organizer of the event."
   }
  ],
  "code": 403,
  "message": "Shared properties can only be changed by the organizer of the event."
 }
}
[gdataCalendar.jsm:575](resource://gdata-provider/legacy/modules/gdataCalendar.jsm)
NS_ERROR_FAILURE: [xpconnect wrapped (nsIClassInfo, calIItemBase, calIEvent, calIInternalShallowCopy)]

**Comment by VeteranCoder** [comment]
I don't have any debug information of my own.  I am definitely have the same symptoms with my editable google calendar.  In my case, I can either create it myself (in Lightning) or receive it in email.  After the meeting passes, I still receive reminders, and dismissing has no effect.  

I noticed a warning badge shows up on the calendar that says it is unable to synchronize right now.

When I delete the event, the reminder does not come back.  To mitigate the annoyance, I turned off the "show reminders for missed events" configuration toggle.  After this, I no longer see the warning badge when I synchronize.  Something is blowing out during that synchronization process, I think.
